server/shadow/X11/x11_shadow.c

```diff
@@ -1176,8 +1182,8 @@ UINT32 x11_shadow_enum_monitors(MONITOR_DEF* monitors, UINT32 maxMonitors)
 		{
 			screens = XineramaQueryScreens(display, &numMonitors);
 
-			if (numMonitors > maxMonitors)
-				numMonitors = maxMonitors;
+			if (numMonitors > (INT64)maxMonitors)
+				numMonitors = (int)maxMonitors;
```

Comment:
Shouldn't these two casts be the same?

Reply:
No, the first one is to check that `numMonitors` is bigger. `maxMonitors` can be larger than `INT_MAX`.
The second one is an assignment to `numMonitors` which we have proven to fit.

Reply:
Ok got it - it's just a bit hard to read ^^